src/AI/JakeMan.java

```diff
@@ -0,0 +1,770 @@
+package AI;
+
+import java.util.*;
+import java.util.Map.Entry;
+
+import AI.CommanderProductionInfo;
+import CommandingOfficers.Commander;
+import CommandingOfficers.CommanderAbility;
+import Engine.*;
+import Engine.UnitActionLifecycles.WaitLifecycle;
+import Terrain.*;
+import Units.*;
+
+
+public class JakeMan extends ModularAI
+{
+  private static class instantiator implements AIMaker
+  {
+    @Override
+    public AIController create(Army army)
+    {
+      return new JakeMan(army);
+    }
+
+    @Override
+    public String getName()
+    {
+      return "JakeMan";
+    }
+
+    @Override
+    public String getDescription()
+    {
+      return
+          "Wants your free dudes.\n" +
+          "High-level design by @Lost&Found#6348, ID 824112693123612692.";
+    }
+  }
+  public static final AIMaker info = new instantiator();
+
+  @Override
+  public AIMaker getAIInfo()
+  {
+    return info;
+  }
+
+  // What % base damage I'll ignore when checking safety
+  private static final int    INDIRECT_THREAT_THRESHHOLD = 7;
+  private static final int    DIRECT_THREAT_THRESHHOLD = 13;
+  // Value to scale the funds damage I deal to something that threatens me
+  private static final double FIRSTSTRIKE_ON_THREAT_WEIGHT = 2.0;
+  private static final int    STAY_UNHURT_BIAS = 1000;
+  private static final int    STAY_ALIVE_BIAS = 2000;
+  // Fraction of the unit to remove from the counter-threat power of my unit type if I'm not attacking
+  private static final double PEACEFUL_SELF_THREAT_RATIO = 1;
+  private static final int    UNIT_HEAL_THRESHHOLD = 6; // HP at which units heal
+  private static final double UNIT_REFUEL_THRESHHOLD = 1.3; // Factor of cost to get to fuel to start worrying about fuel
+  private static final double UNIT_REARM_THRESHHOLD = 0.25; // Fraction of ammo in any weapon below which to consider resupply
+
+  private Map<UnitModel, Map<XYCoord, Double>> unitMapEnemy;
+  private Map<UnitModel, Map<XYCoord, Double>> unitMapFriendly;
+
+  UnitModel infantry;
+  ArrayList<UnitModel> allTanks;
+  UnitModel antiAir;
+  UnitModel copter;
+
+  public JakeMan(Army army)
+  {
+    super(army);
+
+    // look where all vehicles are and what their threat ranges are (yes, mechs are vehicles)
+    // take free dudes that you have more defenders for than them
+    // move your leftover vehicles so they cover the tiles attacking most relevant stuff (contested cities, own units)
+    // move infs as far towards nearest contested stuff as you can without underdefending, cap if possible
+
+    aiPhases = new ArrayList<AIModule>(
+        Arrays.asList(
+            new PowerActivator(army, CommanderAbility.PHASE_TURN_START),
+            new CapChainActuator(army, this),
+            new CaptureFinisher(army, this),
```

Comment:
Doesn't `CapChainActuator` handle capturing cities? Or is that just for early game, and `CaptureFinisher` is needed for subsequent captures?

Reply:
The latter, yeah.
`CapChainActuator` stops doing anything once all the chains are burned.
`CaptureFinisher` is probably not *technically* needed, though you could get weird behavior because infantry would prefer to cap a different city than finish the one they're currently on.

Reply:
Hah. Silly infantry.